AsyncDisplayKit/ASRunLoopQueue.mm

```diff
@@ -235,24 +236,19 @@ - (void)processQueue
     ASProfilingSignpostStart(0, self);
 
     // Snatch the next batch of items.
-    NSUInteger totalNodeCount = _internalQueue.size();
-    for (int i = 0; i < MIN(self.batchSize, totalNodeCount); i++) {
-      id node = _internalQueue[0];
-      itemsToProcess.push_back(node);
-      _internalQueue.pop_front();
-    }
+    auto firstItemToProcess = _internalQueue.cbegin();
+    auto lastItemToProcess = MIN(_internalQueue.cend(), firstItemToProcess + self.batchSize);
+    itemsToProcess = std::vector<id>(firstItemToProcess, lastItemToProcess);
+    _internalQueue.erase(firstItemToProcess, lastItemToProcess);
 
     if (_internalQueue.empty()) {
       isQueueDrained = YES;
     }
   }
 
-  unsigned long numberOfItems = itemsToProcess.size();
-  for (int i = 0; i < numberOfItems; i++) {
-    if (isQueueDrained && i == numberOfItems - 1) {
-      _queueConsumer(itemsToProcess[i], YES);
```

Comment:
The prior logic seemed incorrect to me. If `isQueueDrained == YES` then this will behave the same as the `else` branch. If it's `NO` then it will also behave the same as the else branch.